### PR TITLE
RUST-331 Replace futures_timer with runtime-specific delays

### PR DESCRIPTION
**Pull request by saghm**

### CODE REVIEW

File: src/sdam/message_manager.rs [hunk 1]
```diff
@@ -65,7 +66,7 @@ impl TopologyMessageManager {
 }
 
 async fn wait_for_notification(receiver: &mut Receiver<()>, timeout: Duration) -> bool {
-    let timeout = Delay::new(timeout);
+    let timeout = RUNTIME.delay_for(timeout);
     let message_received = Box::pin(receiver.recv());
 
     match futures::future::select(timeout, message_received).await {
```
Comment (patrickfreed) [inline src/sdam/message_manager.rs]: I think if we replace this `select` with `RUNTIME.timeout`, we can remove the need for `BoxFuture` on `delay_for` and just make it a regular async method.

Comment (saghm) [reply src/sdam/message_manager.rs]: Updated

Comment (patrickfreed) [reply src/sdam/message_manager.rs]: And now `delay_for` can just be `async`, right?

Comment (saghm) [reply src/sdam/message_manager.rs]: Ah, yep. We may need to wrap the return value in a box depending on how we use it, but no need to do that in this case.

**Pull request merged**